EDU-2225-templates-and-integrations-content (#326)
aziontech/docs: src/content/docs/en/pages/marketplace/templates-and-integrations/understand-azion-templates.mdx

- **Configurations setup**: defined according to the use case, to ensure optimal performance and security.
- **Edge function creation**: some cases include the creation of an edge function, providing the arguments with the custom logic and the required dependencies.
- **GitHub repository creation**: this repository contains all the files related to the template, including the edge function, an action to enable continuous deployment workflow, and trackable deployment history. 

Once the template is deployed, you also have full control over customizing all settings at any time you need.

:::caution[warning]
To guarantee the optimal performance of certain templates, some Azion products must be activated in your account and you must enable them separately via RTM. If these products are activated, the execution of the templates could generate usage-related costs. Check the [guides by template](/en/documentation/products/guides/#azion-templates) and the [pricing page](https://www.azion.com/en/pricing/) for more information.
:::

---

## Main configurations and characteristics

> ### User-friendly setting up

Reviewer: ```suggestion
### User-friendly set up
```

> After selecting a template that matches your needs, you must only complete a few steps to quickly deploy the template to the edge through RTM, Azion's configuration interface. With this, you don't need to start from scratch and can have a project running in a few minutes.

Reviewer: ```suggestion
After selecting a template that matches your needs, you only have to complete a few steps to quickly deploy the template to the edge through RTM, Azion's configuration interface. With this, you don't need to start from scratch and can have a project running in a few minutes.
```

After deployment, an edge application, an edge function, and a [GitHub repository](https://github.com/aziontech/azion-samples/tree/dev/templates) are created, including some initial configurations to run the template. This architecture allows an efficient management and continuous deployment workflow.

### Edge capabilities

By using these templates, the applications run on the Azion Edge Platform, taking advantage of a distributed network to accelerate delivery and a use safer environment. Being part of the Azion's platform, you can also benefit from using other [Azion products](/en/documentation/) to enhance your applications. 

### Easy management

You can update and customize your application settings via RTM as well as add a custom domain for users to access your edge application through it. Additionally, the GitHub repository includes a GitHub Action that you're able to activate and enable continuous deployment and integration between your repository and Azion's platform. 

### Third-party integration

> Some templates are integrated with third-party tools to provide a more comprehensive and modern solution. Using this type of template will require compliance with certain requirements such as creating an account or initial setup. In some cases, this integration could generate cost by usage.

Author: ```suggestion
Some templates work with third-party tools to provide a more comprehensive and modern solution. Using this type of template will require compliance with certain requirements such as creating an account or initial setup. In some cases, this integration could generate cost by usage.
```

#### Angular Boilerplate

The Azion Angular Boilerplate provides an automation solution to build an Angular Single-Page Application (SPA) and run it directly on the edge of the network.

<Button href="/en/documentation/products/guides/angular-boilerplate/" text="go to the Angular Boilerplate guide" variant="secondary"> 

</Button>
<br />
<Button href="https://manager.azion.com/build-application/build/choose-template/template/angular/angular-boilerplate" text="deploy this template" variant="secondary"> 

</Button>

#### Astro Boilerplate

The Azion Astro Boilerplate encapsulates and automates several steps, from repository management to edge deployment, to build an Astro Single-Page Application (SPA) that runs directly on the edge.

<Button href="/en/documentation/products/guides/astro-boilerplate/" text="go to the Astro Boilerplate guide" variant="secondary"> 

</Button>
<br />
<Button href="https://manager.azion.com/build-application/build/choose-template/template/astro/astro-boilerplate" text="deploy this template" variant="secondary">

</Button>

> #### Next.js Static Boilerplate
> 
> The Azion Next.js Static Boilerplate enables the creation and deployment of a Next.js Single-Page Application (SPA) directly on the edge of the network.
> 
> <Button href="/en/documentation/products/guides/nextjs-static-boilerplate/" text="go to the Next.js Static Boilerplate guide" variant="secondary"> 
> 
> </Button>
> <br />
> <Button href="#" text="deploy this template" variant="secondary"> 
> 
> </Button>
> 

Author: ```suggestion
```

#### React Boilerplate

> The Azion React Boilerplate helps you build a React Single-Page Application (SPA) directly on the edge By executing the boilerplate, several steps are automated, from repository management to edge deployment.

Reviewer: ```suggestion
The Azion React Boilerplate helps you build a React Single-Page Application (SPA) directly on the edge By executing the boilerplate. Several steps are automated, from repository management to edge deployment.
```

</Button>
<br />
<Button href="https://manager.azion.com/build-application/build/choose-template/template/openai/chatgpt-plugin" text="deploy this template" variant="secondary"> 

</Button>

### Distributed workflows and edge orchestration

Easily manage complex workflows through standardized processes to ensure consistent execution across distributed and serverless environments.

#### QStash EdgeFunction Scheduler

The QStash EdgeFunction Scheduler template sets up and manages a custom edge function, which receives a user-configured schedule and dispatches it through QStash.

<Button href="/en/documentation/products/guides/qstash-edge-function-scheduler/" text="go to the QStash EdgeFunction Scheduler guide" variant="secondary"> 

</Button>
<br />
<Button href="https://manager.azion.com/build-application/build/choose-template/template/upstash/qstash-edgefunction-scheduler" text="deploy this template" variant="secondary"> 

</Button>

### Microservices and API usage

Employ rules and capabilities that help to control and manage incoming traffic and, this way, ensure a more efficient resource allocation and safer environment for your applications.

#### Upstash Rate Limiting

> The Upstash Rate Limiting template helps you to implement rate limiting in a serverless environment, without requiring extensive manual setup. This template includes configurations for limiting requests and window limit and works in conjunction with Azion Edge Functions,

Reviewer: ```suggestion
The Upstash Rate Limiting template helps you to implement rate limiting in a serverless environment, without requiring extensive manual setup. This template includes configurations for limiting requests and window limit and works in conjunction with Azion Edge Functions.
```
